Pod/API/API.swift

```diff
@@ -0,0 +1,129 @@
+    private let manager: Manager
+    /// The target server
+    let server: Server
+    /// The user
+    let username: String
+    /// The MD5 hash of the password
+    private let passwordMD5: String
+    
+    /**
+     Create an API instance for the given server and the given user with the given plaintext password.
+     
+     - parameter server:   The server to target
+     - parameter username: The user
+     - parameter password: The user's plaintext password (only its MD5 hash is stored)
+     
+     - returns: The API instance
+     */
+    public init(server: Server, username: String, password: String, manager: Manager = Manager.sharedInstance) {
+        self.init(server: server, username: username, passwordMD5:(password as NSString).MD5Digest(), manager: manager)
+    }
+    
+    /**
+     Create an API instance for the given server and the given user with the given plaintext password.
+     
+     - parameter server:   The server to target
+     - parameter username: The user
+     - parameter passwordMD5: The MD5 hash of the user's password
+     
+     - returns: The API instance
+     */
+    public init(server: Server, username: String, passwordMD5: String, manager: Manager = Manager.sharedInstance) {
+        self.server = server
+        self.username = username
+        self.passwordMD5 = passwordMD5
+        self.manager = manager
+    }
+    
+    /**
+     Basic internal XMLRPC call wrapper.
+     
+     - parameter method:     The method to call
+     - parameter parameters: The parameters of the call, if any
+     - parameter callback:   The callback to execute with the result of the call
+     */
+    func simpleExecute(method: Method, parameters: [String: Any]? = nil, callback: BaseCallback) {
+        let paramArray: [Any]?
+        if let params = parameters {
+            paramArray = [params]
+        } else {
+            paramArray = nil
+        }
+        self.manager
+            .requestXMLRPC(
+                self.server.xmlrpcEndpoint,
+                methodName: method.methodName,
+                parameters: paramArray)
+            .responseXMLRPC({ response in
+                // Pass along obvious errors.
+                guard let value = response.result.value?[0] where response.result.isSuccess else {
+                    callback(.Failure(.Network(response.result.error!)))
+                    return
+                }
+                
+                // Success.
+                callback(.Success(value))
+            })
+    }
+    
+    /**
+     Internal XMLRPC call handler.  Handles challenge-response authentication.
+     
+     - parameter method:     The method to call
+     - parameter parameters: The parameters of the call, if any
+     - parameter callback:   The callback to execute with the result of the call
+     */
+    func execute(method: Method, parameters: [String: Any]? = nil, callback: BaseCallback) {
+        self.getChallenge({ result in
```

Comment:
trailing closure syntax (I think it makes more sense here than in the previous comment)